exasol_integration_test_docker_environment/lib/base/db_os_executor.py

```diff
@@ -0,0 +1,121 @@
+from abc import abstractmethod
+import fabric
+import docker
+from docker import DockerClient
+from typing import Protocol, runtime_checkable
+from docker.models.containers import Container, ExecResult
+from exasol_integration_test_docker_environment \
+    .lib.base.ssh_access import SshKey
+from exasol_integration_test_docker_environment \
+    .lib.data.database_info import DatabaseInfo
+from exasol_integration_test_docker_environment.lib.docker \
+    import ContextDockerClient
+
+
+class DockerClientFactory:
+    """
+    Create a Docker client.
+    """
+    def __init__(self, timeout: int = 100000):
+        self._timeout = timeout
+
+    def client(self) -> DockerClient:
+        return ContextDockerClient(timeout=self._timeout)
+
+
+@runtime_checkable
+class DbOsExecutor(Protocol):
+    """
+    This class provides an abstraction to execute commands inside a Docker
+    Container.  See concrete implementations in sub-classes
+    ``DockerExecutor`` and ``SshExecutor``.
+    """
+    @abstractmethod
+    def exec(self, cmd: str) -> ExecResult:
+        ...
+
+
+class DockerExecutor(DbOsExecutor):
+    def __init__(self, docker_client: DockerClient, container_name: str):
+        self._client = docker_client
+        self._container_name = container_name
+        self._container = None
+
+    def __enter__(self):
+        self._container = self._client.containers.get(self._container_name)
+        return self
+
+    def __exit__(self, type_, value, traceback):
+        self._container = None
+        self._client.close()
+        self._client = None
```

Comment:
move to close method and call close in `__exit__` `__del__`

Reply:
will be done in next push